### PR TITLE
add wasm32-unknown-unknown support

### PR DESCRIPTION
**Pull request by uuhan**
I make this PR to offer an idea for the wasm support. This is not the best solution even a good solution, but it works in my work.

With this modification, I can run printpdf in my browser by compiled to wasm32-unknown-unknown. 

This is awesome!

### CODE REVIEW

File: src/date.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/// wasm32-unknown-unknown polyfill
+use cfg_if::cfg_if;
+
+cfg_if! {
```
Comment (fschutt) [inline src/date.rs]: You don't need to use `cfg_if`, just do:

```rust
#[cfg(target_arch = "wasm32", target_os = "unknown")]
mod js_sys_data {
    // <js_sys code here>
}

#[cfg(target_arch = "wasm32", target_os = "unknown")]
pub use js_sys_data::OffsetDateTime;
#[cfg(not(target_arch = "wasm32", target_os = "unknown"))]
pub use time::OffsetDateTime;
```

I'd like to do this instead to keep the dependencies to an absolute minimum. The use-case is simple enough to not warrant an external dependency.

File: src/date.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+            #[inline(always)]
+            pub fn now() -> Self {
+                let date = Date::new_0();
+                OffsetDateTime(date)
+            }
+
+            #[inline(always)]
+            pub fn format(&self, format: impl ToString) -> String {
+                "".into()
```
Comment (fschutt) [inline src/date.rs]: Is this intended to be an empty String? Could you use `date.to_iso_string()` instead?

**Review by fschutt — CHANGES_REQUESTED** [review]
Two small issues, otherwise looks good to me

Thanks!

File: src/date.rs (hunk 1 above)
Comment (uuhan) [reply src/date.rs]: Ok, will remove the dep of cfg_if.

File: src/date.rs (hunk 2 above)
Comment (uuhan) [reply src/date.rs]: This function seems not easy to implement due to the `format` arg.
Em, I will try to implement one.

**Comment by eribol** [comment]
In this PR, we cant save the pdf file(save operation does not supported). I tried this PR for my use case but i need how can i use it in browser(like, download the pdf file or open it in new tab)

**Comment by uuhan** [comment]
@eribol 
You can save the pdf file by using URL.createObjectURL(), then   download by attaching it to a <a/> link, for example:

```js
const pdf = lib.pdf(params)
const data = new Blob([pdf], { type: 'application/pdf' })
const file = document.createElement('a')
file.download = 'demo.pdf'
file.href = URL.createObjectURL(data)
file.click()
```

**Pull request merged**